### PR TITLE
[Backport release-1.15] tests(e2e): allow passing options when decoding files

### PR DESCRIPTION
**Pull request by phisco**
Signed-off-by: Philippe Scorsolini <p.scorsolini@gmail.com>
(cherry picked from commit 3a401f8abc7ae6fff7ef310654a9fda9fc347fac)

### Description of your changes

Backport of https://github.com/crossplane/crossplane/pull/5517.

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added or updated unit tests.~
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

File: test/e2e/funcs/feature.go [hunk 1]
```diff
@@ -69,6 +69,8 @@ type onSuccessHandler func(o k8s.Object)
 // AllOf runs the supplied functions in order.
 func AllOf(fns ...features.Func) features.Func {
 	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		t.Helper()
```
Comment (jbw976) [inline test/e2e/funcs/feature.go:72]: all these `t.Helper()` look like they were pre-existing in the cherry-picked https://github.com/phisco/crossplane/commit/3a401f8abc7ae6fff7ef310654a9fda9fc347fac commit - did you have to add these manually? was there another commit you could have cherry-picked to get them instead? 🤔

Comment (phisco) [reply test/e2e/funcs/feature.go:72]: the conflict resolution for some reason tried adding them, coming from https://github.com/crossplane/crossplane/pull/5445, they are a pretty useful addition, so I thought I could just add them too.

Comment (phisco) [reply test/e2e/funcs/feature.go:72]: I can drop them or add a separate commit cherrypicking the original commit if needed

Comment (jbw976) [reply test/e2e/funcs/feature.go:72]: i was mostly curious, don't have a super strong opinion on them. if cherry-picking the original commit is possible and well scoped (i.e. doesn't bring in other things), then that's my preferred approach but not a required approach 😉

Comment (phisco) [reply test/e2e/funcs/feature.go]: just dropped, no need to add the helpers now

**Review by jbw976 — APPROVED** [review]

**Pull request merged**